app/src/main/java/pl/szczodrzynski/edziennik/data/api/edziennik/librus/data/api/LibrusApiTeacherFreeDays.kt

```diff
@@ -36,12 +36,13 @@ class LibrusApiTeacherFreeDays(override val data: DataLibrus,
                 val id = teacherAbsence.getLong("Id") ?: return@forEach
                 val teacherId = teacherAbsence.getJsonObject("Teacher")?.getLong("Id")
                         ?: return@forEach
-                val type = teacherAbsence.getJsonObject("Type").getLong("Id") ?: return@forEach
+                val type = teacherAbsence.getJsonObject("Teacher")?.getLong("Id")
+                    ?: return@forEach
                 val name = data.teacherAbsenceTypes.singleOrNull { it.id == type }?.name
                 val dateFrom = Date.fromY_m_d(teacherAbsence.getString("DateFrom"))
                 val dateTo = Date.fromY_m_d(teacherAbsence.getString("DateTo"))
-                val timeFrom = teacherAbsence.getString("TimeFrom")?.let { Time.fromH_m_s(it) }
-                val timeTo = teacherAbsence.getString("TimeTo")?.let { Time.fromH_m_s(it) }
+                val timeFrom = teacherAbsence.getString("DateFrom")?.let { Time.fromH_m_s(it) }
+                val timeTo = teacherAbsence.getString("DateTo")?.let { Time.fromH_m_s(it) }
```

Comment:
Why is this changed? Date is already retrieved above.

Reply:
TimeFrom and TimeTo are not returned in JSON anymore. App could not display teacher absence when any variable was assigned with non existing object. I could not find way to not declarete the data (leak of knowedge and expirence) so I used this one. Yes it's dirty. What is better solution?

Reply:
If you inspect the resulting object, you will see the time being probably null anyway. Time class cannot parse dates, so it probably simply returns null.